assets/js/base/context/providers/cart-checkout/checkout-processor.ts

```diff
@@ -157,8 +157,25 @@ const CheckoutProcessor = () => {
 		currentRedirectUrl.current = redirectUrl;
 	}, [ billingAddress, shippingAddress, redirectUrl ] );
 
+	// Get shipping rates validation errors from the store.
+	const { shippingRatesValidationError } = useSelect( ( select ) => {
+		const store = select( VALIDATION_STORE_KEY );
+		return {
+			shippingRatesValidationError: store.getValidationError(
+				'shipping-rates-error'
+			),
+		};
+	}, [] );
+
 	const checkValidation = useCallback( () => {
 		if ( hasValidationErrors() ) {
+			// If there is a shipping rates validation error, return the error message to be displayed.
+			if ( shippingRatesValidationError !== undefined ) {
+				return {
+					errorMessage:
+						'Sorry, this order requires a shipping option.',
```

Comment:
This needs to be localised.

Reply:
Ah missed localizing the strings. Thank you for bringing this up. I have added the localization.